### DISCUSSION TITLE
Construct Command / Execute Scratch Buffer

### DISCUSSION THREAD

**Discussion by Randalix** [helix-editor/helix#4808] [Q&A]
I want to pipe an e.g. find command to helix;   construct a command based on the filenames and then execute it.
Is that possible?

**Comment by danillos** [comment]
It could be https://github.com/helix-editor/helix/pull/3393

**Comment by LEI** [comment]
It might depend on your shell and environment, here is what worked for me with the default [bash](https://unix.stackexchange.com/questions/73644/expand-subshell-before-executing-it) readline without copy/paste:
- make helix the default editor with `export VISUAL=hx` (or `EDITOR`),
- prepare the command substitution, e.g; `echo $(find . -type f)`,
- expand the current line with `<Esc><C-e>` (see [here](https://superuser.com/a/247781/724216) for bash aliases),
- use `<C-x><C-e>` to edit the current line in the "visual" editor and execute it on quit.

Alternatively, from within the editor:
- `:insert-output find . -type f`,
- edit and select,
- `:pipe sh`.

Or if you just want to edit the files, `hx $(find . -type f)` would be enough.
